server/activity_manager.go

```diff
@@ -166,15 +167,25 @@ func postWebhookList(webhookCtx webhook.Context, webhookList []*api.ProjectWebho
 	for _, hook := range webhookList {
 		webhookCtx.URL = hook.URL
 		webhookCtx.CreatedTs = time.Now().Unix()
-		if err := webhook.Post(hook.Type, webhookCtx); err != nil {
-			// The external webhook endpoint might be invalid which is out of our code control, so we just emit a warning
-			log.Warn("Failed to post webhook event after changing the issue status",
-				zap.String("webhook_type", hook.Type),
-				zap.String("webhook_name", hook.Name),
-				zap.String("issue_name", issue.Name),
-				zap.String("status", string(issue.Status)),
-				zap.Error(err))
+		suceeded, err := webhook.PostWithRetry(hook.Type, webhookCtx)
+		// Don't log if we succeeded post webhook without retrying.
+		if suceeded && len(multierr.Errors(err)) == 0 {
```

Comment:
Maybe just return if `succeeded`? and just log the warning if `len(multierr.Errors(err)) != 0`